articles/qos.md

```diff
@@ -0,0 +1,104 @@
+
+* Default QoS settings for publishers and subscriptions.
+In order to make the transition from ROS1 to ROS2, exercising a similar network behavior is desirable.
+By default, publishers and subscriptions are reliable in ROS2.
+
+* Services.
+In the same vein as publishers and subscriptions, services are reliable.
+The difference here is that we support some level of durability, so clients can submit requests even before a service is available.
+The durability offered is "transient local", that is, the client's writer will be responsible of persiting the samples until the service can respond to requests. 
+
+* Sensor data.
+For sensor data, in most cases it's more important to receive readings in a timely fashion, rather than ensuring that all of them arrive.
+That is, developers want the latest samples as soon as they are captured, at the expense of maybe losing some.
+
+Profiles allow developers to focus on their applications without worrying about every QoS setting in the DDS specification.
+
+## Integration with existing DDS deployments
+
+Both PrismTech OpenSplice and RTI Connext support loading of QoS policies via an external XML file.
+In environments where DDS is already deployed and also to enable more extensibility other than the offered by the ROS 2.0 and the predefined profiles, ROS 2.0 may provide loading of the QoS settings via the same mechanisms the underlying DDS implementations use.
+However, this mechanism will not be added into the common ROS2 API so as to keep the `rmw` layer transport agnostic and let future developers implement it for other transports (e.g. ZeroMQ, TCPROS, etc.)
+To honor the QoS settings of the system, developers can use the `rmw_qos_profile_system_default` QoS profile which delegates the responsibility of the QoS machinery to the underlying DDS vendor.
+This allows developers to deploy ROS2 applications and use DDS vendor tools to configure the QoS settings.
+
+## Open questions
+
+How should the integration of the QoS machinery with intraprocess communication be like.
+
+## References
+
+* Gordon A. Hunt, [DDS - Advanced Tutorial, Using QoS to Solve Real-World Problems](http://www.omg.org/news/meetings/workshops/RT-2007/00-T5_Hunt-revised.pdf)
+
+* [OpenDDS - QoS Policy Usages](http://www.opendds.org/qosusages.html)
+
+* [OpenDDS - QoS Policy Compliance](http://www.opendds.org/qospolicies.html)
+
+* Angelo Corsaro, [DDS QoS Unleashed](http://www.slideshare.net/Angelo.Corsaro/dds-qos-unleashed)
+
+* Angelo Corsaro, [The DDS Tutorial Part II](http://www.slideshare.net/Angelo.Corsaro/the-dds-tutorial-part-ii)
+
+* [DDS Spec, section 2.2.3](http://www.omg.org/spec/DDS/1.4/PDF/)
+
+* Heidi Schubert, [QoS policies for configuring reliability](https://community.rti.com/content/forum-topic/qos-policies-configuring-reliability)
+
+* [RTI Connext - Comprehensive Summary of QoS Policies](https://community.rti.com/rti-doc/510/ndds.5.1.0/doc/pdf/RTI_CoreLibrariesAndUtilities_QoS_Reference_Guide.pdf)
```

Comment:
Should we link the current version of the document here (https://community.rti.com/static/documentation/connext-dds/5.2.0/doc/manuals/connext_dds/RTI_ConnextDDS_CoreLibraries_QoS_Reference_Guide.pdf)?